cms/toolbar/toolbar.py

```diff
@@ -326,7 +326,7 @@ def get_object_draft_url(self):
             with force_language(self.request_language):
                 try:
                     return self.obj.get_draft_url()
-                except NoReverseMatch:
+                except (NoReverseMatch, AttributeError):
                     try:
                         return self.obj.get_absolute_url()
                     except NoReverseMatch:
```

Comment:
I think it's worth catching the `AttributeError` here also while we're making this change. Models might not have a `get_absolute_url` either.

Reply:
I was wondering about... will do.